src/Bramus/Router/Router.php

```diff
@@ -389,14 +390,23 @@ private function invoke($fn, $params = [])
             }
             // Check if class exists, if not just ignore and check if the class exists on the default namespace
             if (class_exists($controller)) {
-                // First check if is a static method, directly trying to invoke it.
-                // If isn't a valid static method, we will try as a normal method invocation.
-                if (call_user_func_array([new $controller(), $method], $params) === false) {
-                    // Try to call the method as an non-static method. (the if does nothing, only avoids the notice)
-                    if (forward_static_call_array([$controller, $method], $params) === false);
+                // try to invoke controller method by ReflectionMethod 
+                try {
```

Comment:
I don't believe starting a `try/catch` it's the correct thing here. It should be the User to decide how to handle Routing errors, not us.